cpu/stm32/include/periph_cpu.h

```diff
@@ -165,6 +189,38 @@ typedef uint32_t gpio_t;
  */
 #define GPIO_PIN(x, y)      ((GPIOA_BASE + (x << 10)) | y)
 
+/**
+ * @brief   Available GPIO ports
+ */
+enum {
+    PORT_A = 0,             /**< port A */
+    PORT_B = 1,             /**< port B */
+    PORT_C = 2,             /**< port C */
+    PORT_D = 3,             /**< port D */
+    PORT_E = 4,             /**< port E */
+    PORT_F = 5,             /**< port F */
+#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || \
+    defined(CPU_FAM_STM32F3) || defined(CPU_FAM_STM32F4) || \
+    defined(CPU_FAM_STM32F7) || defined(CPU_FAM_STM32L1) || \
+    defined(CPU_FAM_STM32L4)
+    PORT_G = 6,             /**< port G */
+#endif
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F3) || \
+    defined(CPU_FAM_STM32F4) || defined(CPU_FAM_STM32F7) || \
+    defined(CPU_FAM_STM32L0) || defined(CPU_FAM_STM32L1) || \
+    defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32WB)
+    PORT_H = 7,             /**< port H */
+#endif
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
+    defined(CPU_FAM_STM32F7) || defined(CPU_FAM_STM32L4)
+    PORT_I = 8,             /**< port I */
+#endif
+#if defined(CPU_FAM_STM32F7)
+    PORT_J = 9,             /**< port J */
+    PORT_K = 10,            /**< port K */
+#endif
+};
```

Comment:
@aabadie This definition does not seem to be correct. The available ports may also differ within a family. For example, the STM32F410xx has only the ports A...C and H while the STM32F437xx has the ports A...K. Shouldn't it be better defined as follows? The vendor definitions allow to decide which ports are available.
```c
enum {
#ifdef GPIOA
    PORT_A = 0,             /**< port A */
#endif
#ifdef GPIOB
    PORT_B = 0,             /**< port B */
#endif
#ifdef GPIOC
    PORT_C = 0,             /**< port C */
#endif
#ifdef GPIOD
    PORT_D = 0,             /**< port D */
#endif
#ifdef GPIOE
    PORT_E = 0,             /**< port E */
#endif
#ifdef GPIOF
    PORT_F = 0,             /**< port F */
#endif
#ifdef GPIOG
    PORT_G = 0,             /**< port G */
#endif
#ifdef GPIOH
    PORT_H = 0,             /**< port H */
#endif
#ifdef GPIOI
    PORT_I = 0,             /**< port I */
#endif
#ifdef GPIOJ
    PORT_J = 0,             /**< port J */
#endif
#ifdef GPIOK
    PORT_J = 0,             /**< port K */
#endif
};

Reply:
> Shouldn't it be better defined as follows? The vendor definitions allow to decide which ports are available.

That would totally make sense. Mind to open a PR @gschorcht ?